### PR TITLE
Fix/4789  radicale 3 compat

### PR DESCRIPTION
**Pull request by inkhey**
related to #4789 

:warning: DO NOT FORGET TO UPDATE CHANGELOG, migration explain here : https://github.com/tracim/tracim/issues/5479

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [X] Make sure that:
  - ~~if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`~~  -> the settings.md doesn't have a part about radicale config, so not update needed.
  - any migration process required for existing instances is documented : see https://github.com/tracim/tracim/issues/5479
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/applications/agenda/app_factory.py [hunk 1]
```diff
@@ -70,12 +66,13 @@ def create_dir_tree(self, radicale_config: ConfigParser, app_config: CFG):
         os.makedirs(workspace_addressbook_dir, exist_ok=True)
 
     def _parse_additional_radicale_config(
-        self, config: ConfigParser, settings: dict
-    ) -> ConfigParser:
+        self, config: RadicaleConfiguration, settings: dict
+    ) -> dict:
         """
-        Add settings params beginning with
+        get settings params beginning with
```
Comment (Haskaris) [inline backend/tracim_backend/applications/agenda/app_factory.py:72]:
```suggestion
        Get settings params beginning with
```

File: backend/tracim_backend/applications/agenda/app_factory.py [hunk 2]
```diff
@@ -91,15 +88,16 @@ def _parse_additional_radicale_config(
...
             assert sub_main_section == "radicale"
-            if not config.has_section(radicale_section):
-                config.add_section(radicale_section)
-            logger.debug(self, "Override radicale config: {} : {}".format(param_name, value))
-            config.set(radicale_section, radicale_param_config, value)
-        return config
+            if not update_config.get(radicale_section):
+                update_config[radicale_section] = {}
+            logger.debug(
+                self, "Prepare Overriding radicale config: {} : {}".format(param_name, value)
```
Comment (Haskaris) [inline backend/tracim_backend/applications/agenda/app_factory.py:94]:
```suggestion
                self, "Prepare overriding radicale config: {} : {}".format(param_name, value)
```

File: backend/development.ini.sample [hunk 3]
```diff
@@ -730,12 +730,12 @@ webdav.listen = localhost:3030
 ##     param_name = value
 ## ```
 
-caldav.radicale.server.host = localhost:5232
+caldav.radicale.server.hosts = localhost:5232
 ## To make Tracim work, we need to disable auth and rights.
-caldav.radicale.auth.type = None
-caldav.radicale.rights.type = None
+caldav.radicale.auth.type = none
+caldav.radicale.rights.type =  tracim_backend.radicale_plugins.radicale_none_right
```
Comment (Haskaris) [inline backend/development.ini.sample:736]:
```suggestion
caldav.radicale.rights.type = tracim_backend.radicale_plugins.radicale_none_right
```

File: backend/development.ini.sample [hunk 4]
```diff
@@ -730,12 +730,12 @@ webdav.listen = localhost:3030
...
 ## We do not need radicale web ui
-caldav.radicale.web.type = None
+caldav.radicale.web.type = none
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/development.ini.sample]: 2 reasons:
- technical change in radicale3 (don't really know the internal reason, the way caldav deal with config file as change).
- it's an ini file, "None" is not None as in python.

File: backend/development.ini.sample [hunk 5]
```diff
@@ -730,12 +730,12 @@ webdav.listen = localhost:3030
 ##     param_name = value
 ## ```
 
-caldav.radicale.server.host = localhost:5232
+caldav.radicale.server.hosts = localhost:5232
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/development.ini.sample:733]: The answers is yes: https://radicale.org/v3.html#addresses

File: backend/development.ini.sample [hunk 6]
```diff
@@ -730,12 +730,12 @@ webdav.listen = localhost:3030
...
 ## To make Tracim work, we need to disable auth and rights.
-caldav.radicale.auth.type = None
-caldav.radicale.rights.type = None
+caldav.radicale.auth.type = none
+caldav.radicale.rights.type = tracim_backend.radicale_plugins.radicale_none_right
 ## We do not need radicale web ui
-caldav.radicale.web.type = None
+caldav.radicale.web.type = none
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/development.ini.sample:738]: technical choice from radicale, it's in the documentation : https://radicale.org/v3.html#auth

**Review by Haskaris — APPROVED** [review]

File: backend/development.ini.sample (hunk 5 above)
[the remark this answers is not in the document]
Comment (grignards) [reply backend/development.ini.sample:733]: So this is a breaking change?

File: backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._verify_user = self.configuration.get("auth", "type") != "none"
+
+    def authorization(self, user, path):
+        if self._verify_user and not user:
+            return ""
+        sane_path = pathutils.strip_path(path)
+        if "/" not in sane_path:
+            return "RW"
```
Comment (grignards) [inline backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py:15]: Is is logic that `RW` is returned here and `rw` two lines later?

File: backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from radicale import pathutils
```
Comment (grignards) [inline backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py:1]: If I understand right, `radicale_none_right` implements an access right policy which allows everybody.
If this is the case, wouldn't naming it `allow_all_access_rights` be more explicit?

**Review by grignards — CHANGES_REQUESTED** [review]

**Comment by inkhey** [comment]
@grignards yes, this is breaking change, that why changelog update is necessary for this part.

File: backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py (hunk 7 above)
Comment (inkhey) [reply backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py:15]: Yes the meaning is not the same: https://radicale.org/v3.html#authentication-and-rights
RW -> read/right collection
rw -> read/right calendar/addressbook.

File: backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py (hunk 8 above)
Comment (inkhey) [reply backend/tracim_backend/radicale_plugins/radicale_none_right/__init__.py:1]: right

**Pull request merged**